cmd/domain.go

```diff
@@ -69,20 +69,32 @@ ScannerLoop:
 				cli.Errorln("Root Domain Error: ", err)
 			}
 			// Check if root romain has been queried
-			for index, entry := range result {
-				if entry.Name == strings.ToUpper(rootDomain) {
-					//Root domain has already been queried, add to root domain entry "DomainsSearched" .
-					result[index].DomainsSearched = append(result[index].DomainsSearched, strings.ToUpper(i))
-					//Skip query and exit out of current iteration of ScannerLoop.
-					continue ScannerLoop
+			if result != nil {
+
+				for index, entry := range result {
+					if entry.Name == strings.ToUpper(rootDomain) {
+						//Root domain has already been queried, add to root domain entry "DomainsSearched" .
+						result[index].DomainsSearched = append(result[index].DomainsSearched, strings.ToUpper(i))
+						//Skip query and exit out of current iteration of ScannerLoop.
+						continue ScannerLoop
+					}
 				}
-			}
-			if r, e := queryDomain(rootDomain); e != nil {
-				cli.Errorln("Whois lookup error: ", e)
+				if r, e := queryDomain(rootDomain); e != nil {
+					cli.Errorln("Whois lookup error: ", e)
+				} else {
+					result = append(result, r)
+				}
+				time.Sleep(rd)
 			} else {
-				result = append(result, r)
+				if r, e := queryDomain(rootDomain); e != nil {
+					cli.Errorln("Whois lookup error: ", e)
+				} else {
+					result = append(result, r)
+					if i != rootDomain {
```

Comment:
doing this check makes it unclear if the root domain was queried for or not. If the root domain is directly queried it should be included in the domains searched.